### PR TITLE
feat(docs): add boot and advanced topics to core-tutorial

### PR DESCRIPTION
**Pull request by dhmlau**
Originated from https://github.com/strongloop/loopback-next/pull/2939

Adding boot, advanced topics and summary in this PR. 
Once this PR has merged, I will create a new PR for the sidebar.

## Checklist

👉 [Read and sign the CLA (Contributor License Agreement)](https://cla.strongloop.com/agreements/strongloop/loopback-next) 👈

- [ ] `npm test` passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] API Documentation in code was updated
- [ ] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

### CODE REVIEW

File: docs/site/tutorials/core/10-advanced-recipes.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+   ```ts
+   export function globalInterceptor(group?: string) {
+     return bind({tags: {}});
+   }
+   ```
+
+# Create your own injector
+
+```ts
```
Comment (jannyHou) [inline docs/site/tutorials/core/10-advanced-recipes.md]: can we add a link to the example code here? Or makes it clear where do we put the code?

File: docs/site/tutorials/core/10-advanced-recipes.md [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+1. Create a new decorator from scratch by using `DecoratorFactory` from
+   `@loopback/metadata`. See example from
+   [custom-inject-decorator.ts](https://github.com/strongloop/loopback-next/blob/master/examples/context/src/custom-inject-decorator.ts)
+
+2. Create a sugar decorator for an existing decorator.
+
+   ```ts
+   export function globalInterceptor(group?: string) {
+     return bind({tags: {}});
```
Comment (raymondfeng) [inline docs/site/tutorials/core/10-advanced-recipes.md]: `bind({tags: [ContextTags.GLOBAL_INTERCEPTOR]})`

File: docs/site/tutorials/core/10-advanced-recipes.md (hunk 1 above)
Comment (dhmlau) [reply docs/site/tutorials/core/10-advanced-recipes.md]: @jannyHou, i'm trying to find example code in https://github.com/strongloop/loopback-next/tree/master/examples/context but couldn't find any.  Do you know where we can find the example code? Thanks.

Comment (jannyHou) [reply docs/site/tutorials/core/10-advanced-recipes.md]: @dhmlau I think this one https://github.com/strongloop/loopback-next/blob/master/examples/context/src/custom-inject-decorator.ts is a custom injector.

File: docs/site/tutorials/core/10-advanced-recipes.md [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+There are a few advanced usages of the LoopBack core modules that are not
+covered by the Greeter Extension and Greeter Application example.
+
+# Create your own decorator
+
+You can create your own decorator:
+
+1. Create a new decorator from scratch by using `DecoratorFactory` from
+   `@loopback/metadata`. See example from
```
Comment (raymondfeng) [inline docs/site/tutorials/core/10-advanced-recipes.md]: See an example?

File: docs/site/tutorials/core/10-advanced-recipes.md [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/core-tutorial-part10.html
+---
+
+There are a few advanced usages of the LoopBack core modules that are not
+covered by the Greeter Extension and Greeter Application example.
+
+# Create your own decorator
+
+You can create your own decorator:
```
Comment (raymondfeng) [inline docs/site/tutorials/core/10-advanced-recipes.md]: Let's add a link to our main decorator doc page.

File: docs/site/tutorials/core/10-advanced-recipes.md [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+
+2. Create a sugar decorator for an existing decorator.
+
+   ```ts
+   export function globalInterceptor(group?: string) {
+     bind({tags: [ContextTags.GLOBAL_INTERCEPTOR]});
+   }
+   ```
+
+# Create your own injector
```
Comment (raymondfeng) [inline docs/site/tutorials/core/10-advanced-recipes.md]: Let's add a link to our main inject doc page.

File: docs/site/tutorials/core/10-advanced-recipes.md [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+export function env(name: string) {
+  return inject('', {resolve: () => process.env[name]});
+}
+```
+
+For a complete example, see
+https://github.com/strongloop/loopback-next/blob/master/examples/context/src/custom-inject-decorator.ts.
+
+# Class factory to allow parameterized decorations
+
```
Comment (raymondfeng) [inline docs/site/tutorials/core/10-advanced-recipes.md]: Maybe explain why it's needed - as decorations applied on a top-level class cannot be have references to variables.

**Comment by dhmlau** [comment]
@raymondfeng , I've incorporated the comments and please see the commit https://github.com/strongloop/loopback-next/pull/4026/commits/2ce893aa12e431f186e4f0674037fc552d843966.  I will squash the commits once you're good with this.  Thanks.

**Pull request merged**